src/Plugins/SimplnxCore/src/SimplnxCore/Filters/Algorithms/ReadStringDataArray.cpp

```diff
@@ -0,0 +1,98 @@
+#include "ReadStringDataArray.hpp"
+
+#include "simplnx/Common/Range.hpp"
+#include "simplnx/DataStructure/DataArray.hpp"
+#include "simplnx/DataStructure/DataStore.hpp"
+#include "simplnx/DataStructure/Geometry/TriangleGeom.hpp"
+#include "simplnx/Utilities/DataArrayUtilities.hpp"
+#include "simplnx/Utilities/GeometryUtilities.hpp"
+#include "simplnx/Utilities/Parsing/Text/CsvParser.hpp"
+#include "simplnx/Utilities/StringUtilities.hpp"
+
+#include <cstdio>
+#include <utility>
+
+using namespace nx::core;
+
+namespace
+{
+constexpr int32_t k_RBR_FILE_NOT_OPEN = -1000;
+constexpr int32_t k_RBR_READ_EOF = -1030;
+constexpr int32_t k_RBR_READ_ERROR = 1040;
+constexpr int32_t k_RBR_FILE_NOT_EXIST = 1050;
+
+constexpr size_t k_BufferSize = 1024;
+} // End anonymous namespace
+
+namespace nx::core
+{
+ReadStringDataArray::ReadStringDataArray(DataStructure& dataStructure, const IFilter::MessageHandler& mesgHandler, const std::atomic_bool& shouldCancel,
+                                         const ReadStringDataArrayInputValues* inputValues)
+: m_DataStructure(dataStructure)
+, m_MessageHandler(mesgHandler)
+, m_ShouldCancel(shouldCancel)
+, m_InputValues(inputValues)
+{
+}
+
+ReadStringDataArray::~ReadStringDataArray() noexcept = default;
+
+Result<> ReadStringDataArray::operator()()
+{
+  auto data = m_DataStructure.getDataRefAs<StringArray>(m_InputValues->outputArrayPath);
+  char delimiter = nx::core::CsvParser::IndexToDelimiter(m_InputValues->delimiterIndex);
+
+  int32 err;
+  if(!fs::exists(m_InputValues->inputFileValue))
+  {
+    return MakeErrorResult(k_RBR_FILE_NOT_EXIST, fmt::format("Input file does not exist: {}", m_InputValues->inputFileValue.string()));
+  }
+
+  std::ifstream in(m_InputValues->inputFileValue.c_str(), std::ios_base::in | std::ios_base::binary);
+  if(!in.is_open())
+  {
+    return MakeErrorResult(k_RBR_FILE_NOT_OPEN, fmt::format("Could not open file for reading: {}", m_InputValues->inputFileValue.string()));
+  }
+
+  in.imbue(std::locale(std::locale(), new nx::core::CsvParser::DelimiterType(delimiter)));
+
+  std::array<char, k_BufferSize> buf = {};
+  char* buffer = buf.data();
+
+  // Skip some header bytes by just reading those bytes into the pointer knowing that the next
+  // thing we are going to do it over write those bytes with the real data that we are after.
+  for(int i = 0; i < m_InputValues->skipLineCount; i++)
+  {
+    buf.fill(0x00);                                                // Splat Null Chars across the line
```

Comment:
```suggestion
```
Since we are only skipping the lines we don't care about the data in the buffer so setting it to zero is unnecessary. Probably out of scope for this PR but it might make sense to have a dedicated skip line function.

Reply:
replaced with std::getline(in, std::string) which is cleaner.